src/NServiceBus.AcceptanceTests/Sagas/When_saga_is_mapped_to_complex_expression.cs

```diff
@@ -50,7 +50,10 @@ public class TestSaga02 : Saga<TestSagaData02>,
 
                 public Task Handle(OtherMessage message, IMessageHandlerContext context)
                 {
-                    Assert.AreEqual(Context.SagaId, Data.Id, "Existing instance should be found");
+                    if (Context.SagaId != Data.Id)
+                    {
+                        throw new AssertionException("Existing instance should be found");
```

Comment:
I was first doing it but then I realized it is an invariant and the test checks for saga concurrency. I realized I would essentially need to track multiple IDs in a concurrent collection because one single property could be overridden by concurrent started sagas right? Happy to change if you think I'm going over board with my thoughts